searcher.py

```diff
@@ -31,97 +32,88 @@ def __call__(self, tokens):
-    # period – the maximum amount of time (in seconds) between commits.
-    # limit – the maximum number of documents to buffer before committing/between commits.
-    writer = writing.BufferedWriter(ix, period=20, limit=1000)
-    try:
-        # write each file to index
-        # enumerate returns index,value index points too --> index,a[index]
-
-        counter1 = 0
-        for row in df.iterrows():
-            index, data = row
-            writer.add_document(comment_ID=data['name'],
-                                comment_Subreddit=data['subreddit'],
-                                comment_Content=data['body'])
-            counter1 = counter1 + 1
-
-            #             if (counter1 % 100 == 0):
-            #                 print("already indexed:", counter1+1)
-
-    finally:
-        # save the index
-        # print("done indexing")
-        # *** Note *** -> Must explictly call close() on the writer object to release the write lock and makesure uncommited changes are saved
-        writer.close()
-
-
-        # PARSE USER QUERY ****
-
-    # in the query parser --> we pass the DEFAULT field to search and the schema of the index we are searching
-    # NOTE: Users can still specify a search on a different field in the schema via --> <fieldname>: <query>
-    qp = QueryParser("comment_Content", schema=ix.schema)
-
-    # Once you have a QueryParser object, you can call parse() on it to parse a query string into a query object:
-    # default query lang:
-    # If the user doesn’t explicitly specify AND or OR clauses:
-    # by default, the parser treats the words as if they were connected by AND,
-    # meaning all the terms must be present for a document to match
-    # we will change this
-    # to phrase search "<query>" - use quotes
-
-    qp.add_plugin(qparser.GtLtPlugin)
-    # qp.remove_plugin_class(qparser.PhrasePlugin)
-    qp.add_plugin(qparser.PhrasePlugin)
-    query = qp.parse(userQuery)
-    print("\n\n Query: ")
-    print(query)
-    print("\n\n")
-
-    ##IMPLEMENT SEARCHER ****
-    resultsDF = pandas.DataFrame()  # creates a new dataframe that's empty to store the results comment content
-    with ix.searcher(weighting=scoring.BM25F()) as searcher:
-        queryResults = searcher.search(query, limit=None)
-        print("Total Number of Results:", len(queryResults))
-        print("Number of scored and sorted docs in this Results object:", queryResults.scored_length())
-        for result in queryResults:
-            #             print(result)
-            #             print("\n",result['comment_ID'])
-            resultsDF = resultsDF.append(df.loc[df['name'] == result['comment_ID']][['name', 'subreddit', 'body']])
-
-    # print(dataDf.loc[dataDf['body']==comment].index.values[0])
-
-    return resultsDF
\ No newline at end of file
+                      comment_Content=TEXT(analyzer=customWordFilter),
+                      comment_Content_raw=STORED,

```

Comment:
^ new field added to the schema (`comment_Content_raw `) to store everything important in whoosh. We will probably need to add more (vador scores...).